filters/validations.py

```diff
@@ -118,3 +119,73 @@ def fn(value):
                 '<{0}> is not a valid csv of integers'.format(value)
             )
     return fn
+
+
+
+class GenericSeperatedValidator(object):
+    '''
+    Creates list like validator for any voluptuous validation function
+    and any custom string seperator.
+    GenericSeperatedValidator.validation() should be passed to the schema
+
+    >>> CSVofIntegers = GenericSeperatedValidator(int, ',')
+    >>> CSVofIntegers.validation()('1,2,3')
+    [1,2,3]
+
+    >>> WeirdSeperatedValidation = GenericSeperatedValidator(int, '^^')
+    >>> WeirdSeperatedValidation.validation()('1^^2^^3')
+    [1, 2, 3]
+
+    >>> CSVofIntegerLike = GenericSeperatedValidator(IntegerLike(), ',')
+    >>> CSVofIntegerLike.validation()('a,b,c')
+    Traceback (most recent call last):
+        ...
+    voluptuous.error.Invalid: <a,b,c> is not valid set of <IntegerLike>,\
+    Invalid input <a>; expected an integer
+    '''
+
+    def __init__(self, validation_function, seperator=','):
+        if type(seperator) != basestring:
```

Comment:
Use `isinstance(separator, basestring)`. `basestring` is an abstract superclass of `str` and `unicode` so this will return True for all string values.